### ISSUE TITLE
bun add fails in a sub-package of a monorepo workspace

### ISSUE THREAD

**Issue by patlux** [oven-sh/bun#3502]
### What version of Bun is running?

0.5.8, 0.6.0, 0.6.12 (all)

### What platform is your computer?

Darwin 22.5.0 arm64 arm (Doesn't matter)

### What steps can reproduce the bug?

See https://github.com/patlux/bun-workspaces-bug/actions/runs/5438530790/jobs/9889825027

<img width="764" alt="CleanShot 2023-07-02 at 23 05 08@2x" src="https://github.com/oven-sh/bun/assets/4481570/20cd4c6e-46ed-430e-aa98-48958d5266c1">

```sh

git clone https://github.com/patlux/bun-workspaces-bug
cd bun-workspaces-bug
bun install
cd libs/api
bun add redux # or some other package
bun add v0.6.12 (039bbc68)
  🔍 @bun-workspaces-bug/backend [1/1] 
error: package "@bun-workspaces-bug/backend" not found registry.npmjs.org/@bun-workspaces-bug/backend 404
error: @bun-workspaces-bug/backend@1.0.0 failed to resolve
```

If you remove `@bun-workspaces-bug/backend` from `libs/api/package.json` it will work.

### What is the expected behavior?

Should work

### What do you see instead?

```sh
bun add v0.6.12 (039bbc68)
  🔍 @bun-workspaces-bug/backend [1/1] 
error: package "@bun-workspaces-bug/backend" not found registry.npmjs.org/@bun-workspaces-bug/backend 404
error: @bun-workspaces-bug/backend@1.0.0 failed to resolve
```

### Additional information

It only works, if the sub-package doesn't have a dependency of the workspace.

`bun install` works without any issue in the root folder of the monorepo.

**Comment by jraoult** [comment]
I was about to file an issue but I think this is the same bug even if the message is now different. This is what shows up when I try to add `koa` to the `server` workspace of the following repo: https://github.com/DerekZiemba/yarn-V2-workspaces-simple-monorepo:

```bash
~/Desktop/yarn-V2-workspaces-simple-monorepo/packages/server yarn2*
❯ bun add koa
bun add v0.7.3 (a9b3d583)


error: workspace dependency "@example/example-1" not found

Searched in "./*"

Workspace documentation: https://bun.sh/docs/install/workspaces
```

**Comment by Jarred-Sumner** [comment]
Hmm it sounds like https://github.com/oven-sh/bun/issues/3985 is not fully resolved

**Comment by jraoult** [comment]
@Jarred-Sumner yep, that's the other one I had my eyes on because of the error message similarities. But since it was marked as resolved and because my issue is only happening when adding a npm module to a workspace I figure this one was more accurate.

**Comment by Jarred-Sumner** [comment]
Confirming that I am able to reproduce this in Bun v0.7.3. 

@alexlamsl can you take a look?

**Comment by danieldunderfelt** [comment]
This is still a problem for me in Bun 1.0. Using "bun add", or "bun install" if there is anything to install, with a lockfile present always fails when it tries to fetch a workspace package from npm. I have tried all configurations of workspace packages and how they appear in dependencies of the other packages, but no luck.

**Comment by paulleonartcalvo** [comment]
> This is still a problem for me in Bun 1.0. Using "bun add", or "bun install" if there is anything to install, with a lockfile present always fails when it tries to fetch a workspace package from npm. I have tried all configurations of workspace packages and how they appear in dependencies of the other packages, but no luck.

Same, I can install at the root, but adding a package from within any of the workspaces results in `workspace dependency myDep not found`

**Comment by poyrazbirbil** [comment]
same here!

**Comment by dangvanthanh** [comment]
I have same issue when working menorepo workspace: https://github.com/dangvanthanh/paga-ui

I try remove `lock.bun` then install:

<img width="856" alt="Screenshot 2023-09-22 at 5 20 08 PM" src="https://github.com/oven-sh/bun/assets/2674850/f8fee081-9d22-4a6f-bbe0-ffd3804fba6f">

But when I add new package in `package/web`. It show this error:

<img width="861" alt="Screenshot 2023-09-22 at 5 20 32 PM" src="https://github.com/oven-sh/bun/assets/2674850/9df6d9c7-a05b-442d-969e-edfe7392bd03">

**Comment by arunrdd** [comment]
`bun add remote-package` works without errors when using a relative path for a workspace dependency, like so in `package.json` of `package-a`:
```
'package-b': 'workspace:../../packages/package-b'
```
However, `bun install` in the root directory of the monorepo only works when the dependency is listed like below in `package.json`:
```
'package-b': 'workspace:*'
```
I am temporarily switching between two to add any additional remote packages as dependencies before this issue gets fixed.

**Comment by sgtsquiggs** [comment]
I can still reproduce this issue today with bun 1.0.35

Every time I add a dependency to a workspace package I have to remove all workspace dependencies from all workspace packages, bun install, then manually add all the workspace dependencies back.

**Comment by SiebeBaree** [comment]
It's also still an issue with bun 1.1.3

**Comment by patlux** [comment]
It should work since 1.0.10 at least.

![CleanShot 2024-04-10 at 16 17 36](https://github.com/oven-sh/bun/assets/4481570/8e7acab7-33f6-4c53-a445-4f0723efde82)

See https://github.com/patlux/bun-workspaces-bug/actions/runs/8632776332

**Comment by SiebeBaree** [comment]
I'm getting the following error:
![image](https://github.com/oven-sh/bun/assets/33906209/4f828b09-7b6d-4dfe-bea6-45d235ad2058)

And here is my .eslintrc.cjs file:
![image](https://github.com/oven-sh/bun/assets/33906209/056af6e3-6f06-4a38-a2b6-8b4d474c8ade)

But I ran `bun install` multiple times and removed node_modules.

I added `"@repo/config": "*",` to devDependencies in my package.json and here is my package.json inside packages/config
![image](https://github.com/oven-sh/bun/assets/33906209/3e05db8f-79b8-4041-b999-5045d7943c97)

My packages/config/eslint/bot.js file:
![image](https://github.com/oven-sh/bun/assets/33906209/e1b5afe6-92fb-444a-af4e-614385c3e1d6)
![image](https://github.com/oven-sh/bun/assets/33906209/a2462853-2c9c-4ffd-b343-a59f84199f68)

**Comment by SiebeBaree** [comment]
![image](https://github.com/oven-sh/bun/assets/33906209/068248f4-1f5e-427f-908d-0b561b623e4b)

**Comment by dylan-conway** [comment]
This is fixed in latest bun, most likely since `v1.1.9` with #11177
